src/crypto/test/digest_bench.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "crypto/hash.h"
+
+#define PICOBENCH_IMPLEMENT_WITH_MAIN
+#include <picobench/picobench.hpp>
+
+static void evercrypt_sha256(picobench::state& s)
+{
+  std::vector<uint8_t> v(s.iterations());
+  for (size_t i = 0; i < v.size(); ++i)
+  {
+    v.data()[i] = rand();
+  }
+
+  crypto::Sha256Hash h;
+
+  s.start_timer();
+  for (size_t i = 0; i < 1000; ++i)
+  {
+    crypto::Sha256Hash::evercrypt_sha256(v, h.h.data());
+  }
+  s.stop_timer();
+}
+
+static void mbedtls_sha256(picobench::state& s)
+{
+  std::vector<uint8_t> v(s.iterations());
+  for (size_t i = 0; i < v.size(); ++i)
+  {
+    v.data()[i] = rand();
+  }
+
+  crypto::Sha256Hash h;
+
+  s.start_timer();
+  for (size_t i = 0; i < 1000; ++i)
+  {
+    crypto::Sha256Hash::mbedtls_sha256(v, h.h.data());
```

Comment:
Nit: We could combine these so we have less duplicated code to keep in sync, but its a little awkward. Something like:

```
template<HashImpl IMPL>
static void digest_bench(...)
{
...
if constexpr (IMPL == HashImpl::mbedtls)
{
  crypto::Sha256Hash::mbedtls_sha256(v, h.h.data());
}
else if constexpr (IMPL == HashImpl::evercrypt)
{
  crypto::Sha256Hash::evercrypt_sha256(v, h.h.data());
}
...
}

auto mbedtls_sha256 = digest_bench<HashImpl::mbedtls>;
auto evercrypt_sha256 = digest_bench<HashImpl::evercrypt>;
```

Reply:
Yeah, I almost did it, and then I realised it was actually just as much code as duplicating. I don't mind.